float32 distance_to_obj
